internal_worker.go

```diff
@@ -348,43 +348,59 @@ func (th *hostEnvImpl) AddActivityRegistrationInterceptor(i interceptorFn) {
 	}
 }
 
-func (th *hostEnvImpl) RegisterWorkflow(af interface{}) error {
+func (th *hostEnvImpl) RegisterWorkflow(af interface{}, options RegisterWorkflowOptions) error {
 	// Validate that it is a function
 	fnType := reflect.TypeOf(af)
 	if err := validateFnFormat(fnType, true); err != nil {
 		return err
 	}
-	// Check if already registered
-	fnName := getFunctionName(af)
-	if _, ok := th.getWorkflowFn(fnName); ok {
-		return fmt.Errorf("workflow type \"%v\" is already registered", fnName)
+	names := make([]string, 0, 2)
+	names = append(names, getFunctionName(af))
```

Comment:
I think when name is present in the options there is no need to register workflow under fn name.
If somebody really needs (for example for migration) they can just call Register twice with the same function.